src/jobs/database_schema_report.yml

```diff
@@ -0,0 +1,50 @@
+---
+description: >
+  Job to generate a schema report for a postgres database, using schemaspy. Runs against postgres in docker
+executor:
+  name: java
+  tag: "21.0"
+parameters:
+  host:
+    type: string
+    default: localhost
+    description: Host name of the database
+  port:
+    type: string
+    default: "5432"
+    description: Port number of the database
+  schema:
+    type: string
+    description: Schema name
+  user:
+    type: string
+    default: root
+    description: Database user name
+  password:
+    type: string
+    default: dev
+    description: Database password
+steps:
+  - run:
+      name: Install schemaspy
+      command: curl -L https://github.com/schemaspy/schemaspy/releases/download/v6.2.4/schemaspy-6.2.4.jar --output /tmp/schemaspy.jar
+  - run:
+      name: Install postgres jdbc driver
+      command: curl -L https://jdbc.postgresql.org/download/postgresql-42.7.3.jar --output /tmp/postgres-driver.jar
```

Comment:
would be better not to hard code versions in jobs

Reply:
We can do but then users will have to manage this themselves. Is it not simpler to manage this via the job? 

We could end up with the state of affairs we have with the localstack executor where we have to support really old versions because people are passing them in. These warnings could be avoided but there are a lot of projects still on version 0.x of localstack, which uses these properties.

```2024-05-10T13:29:54.006  WARN --- [  MainThread] localstack.deprecations    : DEFAULT_REGION is deprecated (since 0.12.7) and will be removed in upcoming releases of LocalStack! LocalStack now has full multi-region support. This option has no effect. Please remove it from your configuration.
2024-05-10T13:29:54.006  WARN --- [  MainThread] localstack.deprecations    : DATA_DIR is deprecated (since 1.0.0) and will be removed in upcoming releases of LocalStack! Please use PERSISTENCE instead. The state will be stored in your LocalStack volume in the state/ directory.
2024-05-10T13:29:54.006  WARN --- [  MainThread] localstack.deprecations    : TMPDIR is deprecated (since 1.0.0) and will be removed in upcoming releases of LocalStack! Please migrate to the new filesystem layout (introduced with v1.0).
2024-05-10T13:29:54.007  WARN --- [  MainThread] localstack.deprecations    : LAMBDA_EXECUTOR is deprecated (since 2.0.0) and will be removed in upcoming releases of LocalStack! This configuration is obsolete with the new lambda provider https://docs.localstack.cloud/user-guide/aws/lambda/#migrating-to-lambda-v2```

Reply:
That relies on someone checking this job manually on a weekly basis or so to ensure that we are up to date.  If there are newer versions then a PR needs to be raised and a new orb released generated.  The other problem with the orb doing this is that it will need to ensure that everyone else's projects work after the update - either that or release a major version of the orb each time.
Historically that hasn't happened - there is normally very little involvement in community projects by the majority of HMPPS developers.
Regarding localstack - the warnings are only there because no-one has upgraded the orb to remove the properties.  It is as simple as removing them and then releasing a major version of the orb.

Reply:
Added versions into parameters